app/controllers/tickets_controller.rb

```diff
@@ -65,10 +65,10 @@ def show
   def update_status
     ticket = Ticket.find(params.require(:ticket_id))
     ticket_status = params.require(:ticket_status)
-    acting_stakeholder_id = params.require(:acting_stakeholder_id)
+    acting_stakeholder = TicketStakeholder.find(params.require(:acting_stakeholder_id))
 
-    return head :unauthorized unless ticket.action_allowed?(:update_status, current_user)
-    return head :bad_request unless ticket.user_stakeholders(current_user).map(&:id).include?(acting_stakeholder_id)
+    return head :bad_request unless ticket.user_stakeholders(current_user).map(&:id).include?(acting_stakeholder.id)
```

Comment:
(see other comment about potentially using `.pluck(:id)`, feel free to just discard this comment if the other line doesn't work)

Reply:
Changed that too